canvas/models.py

```diff
@@ -269,6 +269,14 @@ def cannot_access_event_yet(self, user):
     def is_exam_and_open(self):
         return self.is_exam and self.is_open
 
+    def clone(self, course):
+        course_clone = self
```

Comment:
Here you should iterate over the questions and clone them. The clone method should work independently.